### ISSUE TITLE
[SPIKE] aws_iam_user and aws_iam_access_key for Recommendation 1.4

### ISSUE THREAD

**Issue by chris-redekop** [chef/inspec-aws#37]
We have user and access-key resources.  What else do we need before we can implement a test for Recommendation 1.4?

**Comment by mtjandra** [comment]
Building off https://github.com/chef/inspec-aws/pull/44/files I think we want to implement two methods to:
1. Get the last rotated date for an access key
2. Check if access key has been used since last rotation

**Proposal:** Create to issues to implement the two methods above such that the end result looks something like:
```
aws_iam_user(some_user).access_keys.each { |access_key|
  if access_key.active?
    describe access_key do
      its('last_rotated_date') { should be > Time.now - 90 * 86400 }
      it('should be_used_since_last_rotation')
    end
  end
}
```

**Comment by mtjandra** [comment]
Above I referenced using the credentials report which should not be used, see https://github.com/chef/inspec-aws/issues/38. If I understand correctly we should be able to satisfy 1.4 with the information of when the access key was last created/updated and when it was last used in relation to when the key was last created/updated. This is based on the definition of a rotation from http://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_getting-report.html.  Oddly it seems to suggest you can toggle the status of your access key `Active -> Inactive -> Active` and this counts as a rotation. Also AWS sources suggest rotating keys by:

1. Create new key
2. Update apps
3. Deactivate old key
4. Delete old key

TLDR; The above can be achieved without the use of the credential report. Someone may want to validate the definition of rotation, and consistency between credential report and access key meta data.

**Comment by chris-redekop** [comment]
I think you're on the right track, @mtjandra : we can use the create date.  Discussing with @Rugbyte , I think we can look for active keys that were created > 90 days ago.  (This would not treat a deactiviate/activate cycle, per se, as a rotation.)

What do you think?

**Comment by chris-redekop** [comment]
@mtjandra , just waiting for your response here before we close this one.   Objections?

**Comment by mtjandra** [comment]
Yup sounds good! Can be closed and implemented. I guess I misinterpreted `Active -> Inactive -> Active` as being a valid rotation, or perhaps we are not following the definition of rotation that AWS provides. `The date and time, in ISO 8601 date-time format, when the user's access key was created or **last changed**.`

**Comment by chris-redekop** [comment]
Added #64 to cover this work.
